pkg/tfvars/openstack/openstack.go

```diff
@@ -41,20 +41,28 @@ type config struct {
 	AdditionalSecurityGroupIDs []string `json:"openstack_master_extra_sg_ids,omitempty"`
 	MachinesSubnet             string   `json:"openstack_machines_subnet_id,omitempty"`
 	MachinesNetwork            string   `json:"openstack_machines_network_id,omitempty"`
+	MasterAvailibilityZones    []string `json:"openstack_master_availibility_zones,omitempty"`
 }
 
 // TFVars generates OpenStack-specific Terraform variables.
-func TFVars(masterConfig *v1alpha1.OpenstackProviderSpec, cloud string, externalNetwork string, externalDNS []string, lbFloatingIP string, apiVIP string, ingressVIP string, baseImage string, infraID string, userCA string, bootstrapIgn string, mpool *types_openstack.MachinePool, machinesSubnet string) ([]byte, error) {
+func TFVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, externalNetwork string, externalDNS []string, lbFloatingIP string, apiVIP string, ingressVIP string, baseImage string, infraID string, userCA string, bootstrapIgn string, mpool *types_openstack.MachinePool, machinesSubnet string) ([]byte, error) {
+	cloud := masterConfigs[0].CloudName
+
+	var zones []string
+	for _, config := range masterConfigs {
+		zones = append(zones, config.AvailabilityZone)
```

Comment:
This will cause duplicate entries if you have more master nodes than availability zones. We should check if the element is already there before appending it to the slice.